macosx/Controller.mm

```diff
@@ -5246,6 +5247,21 @@ - (void)setWindowSizeToFit
     }
 }
 
+- (CGFloat)calculateScrollViewHeightWithDockAdjustment
+{
+    CGFloat height = [self scrollViewHeight];
+
+    // Get the main screen's visible frame
+    NSScreen* screen = self.fWindow.screen;
+    if (screen)
+    {
+        NSRect visibleFrame = screen.visibleFrame; // This frame respects the Dock and menu bar
```

Comment:
I prefer comments on their own line.
Yes, I did write myself trailing comments (like in b1f6fbc), but it was to distinguish a comment on the _if condition_ versus a comment on the _whole block content_.